### ISSUE TITLE
Delta stepping with custom vertices fails

### ISSUE THREAD

**Issue by d-michail** [jgrapht/jgrapht#1074]
```
 * JGraphT version: 1.5.1
 * Java version (java -version)/platform:  11
```

**Issue**

I was trying to execute `DeltaSteppingShortestPath` with a custom vertex which does not implement comparable. I am getting a `CustomVertex  cannot be cast to java.lang.Comparable`.  The use of `ConcurrentSkipListSet` in the buckets forces the user to use comparable vertices. 

My first guess is that this can be avoided. If not, we should at least allow the user to provide an external comparator.

**Steps to reproduce (small coding example)**

Add the following code in `DeltaSteppingShortestPathTest`: 

```
    public class MyVertex  { 
        public MyVertex() { 
        }
    }
    
    @Test
    public void testCustomVertex()
    {
        Graph<MyVertex, DefaultEdge> graph =
            new DirectedWeightedPseudograph<>(DefaultEdge.class);
        
        MyVertex v0 = new MyVertex();
        MyVertex v1 = new MyVertex();
        graph.addVertex(v0);
        graph.addVertex(v1);
        graph.addEdge(v0, v1);

        new DeltaSteppingShortestPath<>(graph, executor).getPaths(v0);
    }
```

**Expected behaviour**

No exception.

**Other information**

**Comment by d-michail** [comment]
@SChudakov Could you look into this?

**Comment by SChudakov** [comment]
Resolved in #1094.
